src/components/organisms/Room/Participant.tsx

```diff
@@ -97,37 +111,29 @@ const Participant: React.FC<React.PropsWithChildren<ParticipantProps>> = ({
     }
   }, [audioTracks]);
 
-  useEffect(() => {
-    if (muted) {
-      const audioTrack = audioTracks[0];
-      if (audioTrack) {
-        audioTrack.detach();
+  const changeAudioState = useCallback(
+    (enable: boolean) => {
+      if (!enable) {
+        audioTracks?.[0]?.detach();
+      } else {
+        const audioTrack = audioTracks[0];
+        if (audioTrack && audioRef.current) {
+          audioTrack.attach(audioRef.current);
+        }
       }
-    } else {
-      const audioTrack = audioTracks[0];
-      if (audioTrack && audioRef.current) {
-        audioTrack.attach(audioRef.current);
-      }
-    }
-  }, [participant, muted, audioTracks]);
-
-  const videos = useMemo(
-    () => (
-      <>
-        <video
-          ref={videoRef}
-          autoPlay={true}
-          className={profileData?.mirrorVideo ? "mirrored" : ""}
-        />
-        <audio ref={audioRef} autoPlay={true} />
-      </>
-    ),
-    [profileData]
+    },
+    [audioTracks]
   );
 
   return (
     <div className={`col participant ${bartender ? "bartender" : ""}`}>
-      {videos}
+      {!videoEnabled && <div className="participant--video-disabled" />}
+      <video
+        ref={videoRef}
+        autoPlay={true}
+        className={profileData?.mirrorVideo ? "mirrored" : ""}
+      />
+      <audio ref={audioRef} autoPlay={true} />
```

Comment:
why was that removed from useMemo?

Reply:
restored in https://github.com/sparkletown/sparkle/pull/2221/commits/c8ea9c6fae402dd1cb702dba9ecf856e61232663